parallel-install/pkg/postuninstaller/postuninstaller.go

```diff
@@ -0,0 +1,161 @@
+package postuninstaller
+
+import (
+	"context"
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/config"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/logger"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/preinstaller"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/client-go/dynamic"
+)
+
+// Config defines configuration values for the PostUninstaller.
+type Config struct {
+	Log                      logger.Interface        //Logger to be used
+	KubeconfigSource         config.KubeconfigSource //KubeconfigSource to be used
+	InstallationResourcePath string
+}
+
+//go:generate mockery --name PostUninstaller
+
+// PostUninstaller removes leftover resources from k8s cluster, added during Kyma installation.
+type PostUninstaller struct {
+	cfg             Config
+	dynamicClient   dynamic.Interface
+	resourceManager preinstaller.ResourceManager
+	retryOptions    []retry.Option
+}
+
+// Output contains lists of Deleted and NotDeleted resources during PostUninstaller uninstallation.
+type Output struct {
+	// Installed files during PreInstaller installation.
+	Deleted []string
+	// NotInstalled files during PreInstaller installation.
+	NotDeleted []string
+}
+
+// NewPostUninstaller creates a new instance of PostUninstaller.
+func NewPostUninstaller(cfg Config, retryOptions []retry.Option) (*PostUninstaller, error) {
+	restConfig, err := config.RestConfig(cfg.KubeconfigSource)
+	if err != nil {
+		return nil, err
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceManager, err := preinstaller.NewDefaultResourceManager(cfg.KubeconfigSource, cfg.Log, retryOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PostUninstaller{
+		cfg:             cfg,
+		dynamicClient:   dynamicClient,
+		resourceManager: resourceManager,
+		retryOptions:    retryOptions,
+	}, nil
+}
+
+// UninstallCRDs that belong to Kyma from a k8s cluster.
```

Comment:
I'm personally not a fan of this kind of comments. It does not say much more than the func declaration. I'd love to see comments that answers more to "why? how?" questions.

Reply:
I will extend this one with some implementation details of how it does it internally. 👍